crates/bevy_ui/src/focus.rs

```diff
@@ -161,52 +168,49 @@ pub fn ui_focus_system(
                 };
 
                 if contains_cursor {
-                    Some((entity, focus_policy, interaction, FloatOrd(position.z)))
-                } else {
-                    if let Some(mut interaction) = interaction {
-                        if *interaction == Interaction::Hovered
-                            || (cursor_position.is_none() && *interaction != Interaction::None)
-                        {
-                            *interaction = Interaction::None;
-                        }
+                    return Some(*entity);
+                } else if let Some(mut interaction) = node.interaction {
+                    if *interaction == Interaction::Hovered
+                        || (cursor_position.is_none() && *interaction != Interaction::None)
+                    {
+                        *interaction = Interaction::None;
                     }
-                    None
                 }
-            },
-        )
-        .collect::<Vec<_>>();
-
-    moused_over_z_sorted_nodes.sort_by_key(|(_, _, _, z)| -*z);
+            }
+            None
+        })
+        .collect::<Vec<Entity>>();
 
-    let mut moused_over_z_sorted_nodes = moused_over_z_sorted_nodes.into_iter();
     // set Clicked or Hovered on top nodes
-    for (entity, focus_policy, interaction, _) in moused_over_z_sorted_nodes.by_ref() {
-        if let Some(mut interaction) = interaction {
+    let mut iter = node_query.iter_many_mut(&moused_over_nodes);
+    while let Some(node) = iter.fetch_next() {
+        if let Some(mut interaction) = node.interaction {
             if mouse_clicked {
                 // only consider nodes with Interaction "clickable"
                 if *interaction != Interaction::Clicked {
                     *interaction = Interaction::Clicked;
                     // if the mouse was simultaneously released, reset this Interaction in the next
                     // frame
                     if mouse_released {
-                        state.entities_to_reset.push(entity);
+                        state.entities_to_reset.push(node.entity);
                     }
                 }
             } else if *interaction == Interaction::None {
                 *interaction = Interaction::Hovered;
             }
         }
 
-        match focus_policy.cloned().unwrap_or(FocusPolicy::Block) {
+        match node.focus_policy.cloned().unwrap_or(FocusPolicy::Block) {
```

Comment:
Can we just make `FocusPolicy` a `Copy` type instead?

Reply:
It's already copy. `cloned` was used here to transform `Option<&FocusPolicy>` to `Option<FocusPolicy>`. I went the other way around and simplified with `node.focus_policy.unwrap_or(&FocusPolicy::Block)`.